### DISCUSSION TITLE
Name clash for environment variables

### DISCUSSION THREAD

**Discussion by sberlin** [prodrigestivill/docker-postgres-backup-local#112] [Q&A]
The used environment variables, mostly prefixed with `POSTGRES_` lead to a collision in my setup. I used your image in OpenShift to back up an instance (a.k.a. deployment) named 'postgres'. OpenShift uses the deployment name to inject environment variables similar to the one already read by the script: `POSTGRES_PORT_5432_TCP_ADDR`. This lead to the variable `POSTGRES_PORT` being set to `tcp://x.x.x.x:5432` by OpenShift as soon as I configured the custom variable `POSTGRES_HOST`. This can be fixed with an additional switch checking the environment variables, but I think the underlying issue is the missing global namespace for custom environment variables. Do you agree or do you like to avoid breaking changes altogether? I won't make an PR with the switch as this is really an edge case, an error in usage even so. Instead the variable `POSTGRES_PORT_5432_TCP_ADDR` should be documented, I think. Effectively the implementation forces OpenShift users having a deployment named 'postgres' to unset `POSTGRES_HOST`.

**Comment by prodrigestivill** [comment]
Added variables description to README.
Not sure what variables are set by OpenShift in your deployment.
And not sure what exactly do you need.

Do you need different prefix for all variables?

**Comment by sberlin** [comment]
I'm fine with my current setup and think amending the documentation should suffice.
It would be more convenient for users to have a namespace for variable names to avoid unexpected behavior like I observed. Fortunately at this point no major problems arise in use cases I can think of. Thus I will close this issue.

Thank you.
